recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -63,39 +72,45 @@ def _configure_cmake(self):
         self._cmake.definitions["PAHO_BUILD_ASYNC"] = self.options.asynchronous
         self._cmake.definitions["PAHO_BUILD_STATIC"] = not self.options.shared
         self._cmake.definitions["PAHO_BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["PAHO_BUILD_SAMPLES"] = self.options.samples
+        self._cmake.definitions["PAHO_BUILD_SAMPLES"] = False
         self._cmake.definitions["PAHO_WITH_SSL"] = self.options.ssl
         if self.options.ssl:
-            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath
-            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath
+            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
         self._cmake.configure()
         return self._cmake
 
-    def build(self):
+    def _patch_source(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+        if not self.options.get_safe("fPIC", True):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "POSITION_INDEPENDENT_CODE ON", "")
+
+    def build(self):
+        self._patch_source()
         cmake = self._configure_cmake()
-        cmake.build()
+        cmake.build(target=self._cmake_target)
 
     def package(self):
         self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
-        if self.version in ['1.3.0', '1.3.1']:
-            eplfile = "epl-v10"
-        else:
-            eplfile = "epl-v20" # EPL changed to V2
-        self.copy(eplfile, src=self._source_subfolder, dst="licenses")
+        self.copy(self._epl_file, src=self._source_subfolder, dst="licenses")
         self.copy("notice.html", src=self._source_subfolder, dst="licenses")
-        cmake = self._configure_cmake()
-        cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        # Manually copy since the CMake installs everything
+        self.copy("*.h", src=self._source_subfolder, dst="include", keep_path=False)
+        self.copy(os.path.join("lib", "*{}.*".format(self._lib_target)), dst="lib", keep_path=False)
+        self.copy(os.path.join("bin", "*{}.*".format(self._lib_target)), dst="bin", keep_path=False)
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.pdb")
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
```

Comment:
Honestly, I prefer the previous revision regarding installation. Here it installs too many headers.
In master, public headers are listed here: https://github.com/eclipse/paho.mqtt.c/blob/64a5ff3c3b71fe019353aeacaebc66a3cf4f3461/src/CMakeLists.txt#L185-L186

Reply:
At least, try to filer headers: `self.copy("MQTT*.h", dst="include", src=os.path.join(self._source_subfolder, "src"))`

Not perfect, but still better.

Reply:
I agree it's not perfect, but version bumps were a nightmare... being able to have new versions **and** understand what the patch does I think is a good compromise.